src/mngt/ocf_mngt_cache.c

```diff
@@ -960,14 +957,44 @@ static void _ocf_mngt_attach_load_properties(ocf_pipeline_t pipeline,
 	context->metadata.dirty_flushed = DIRTY_FLUSHED;
 	context->metadata.line_size = context->cfg.cache_line_size;
 
+	ocf_pipeline_next(pipeline);
+}
+
+static void _ocf_mngt_attach_read_properties(ocf_pipeline_t pipeline,
+		void *priv, ocf_pipeline_arg_t arg)
+{
+	struct ocf_cache_attach_context *context = priv;
+	ocf_cache_t cache = context->cache;
+
 	if (context->cfg.force)
-		OCF_PL_NEXT_RET(context->pipeline);
+		OCF_PL_NEXT_RET(pipeline);
+
+	if (cache->metadata.is_volatile)
+		OCF_PL_NEXT_RET(pipeline);
+
+	ocf_metadata_load_properties(&cache->device->volume,
+			_ocf_mngt_attach_read_properties_end, context);
+}
 
-	if (cache->device->init_mode == ocf_init_mode_metadata_volatile)
-		OCF_PL_NEXT_RET(context->pipeline);
+static void _ocf_mngt_load_read_properties(ocf_pipeline_t pipeline,
+		void *priv, ocf_pipeline_arg_t arg)
+{
+	struct ocf_cache_attach_context *context = priv;
+	ocf_cache_t cache = context->cache;
 
 	ocf_metadata_load_properties(&cache->device->volume,
-			_ocf_mngt_attach_load_properties_end, context);
+			_ocf_mngt_load_read_properties_end, context);
+}
+
+static void _ocf_mngt_load_check_metadata(ocf_pipeline_t pipeline,
+		void *priv, ocf_pipeline_arg_t arg)
+{
+	struct ocf_cache_attach_context *context = priv;
+
+	if (context->metadata.status)
+		OCF_PL_FINISH_RET(pipeline, -OCF_ERR_START_CACHE_FAIL);
```

Comment:
since we are reworking this area anyway: maybe we can do better then -OCF_ERR_START_CACHE_FAIL? I suppose context->metadata.status already contains meaningfull error (maybe good to verify this).